### DISCUSSION TITLE
How to skip over redundant data

### DISCUSSION THREAD

**Discussion by Zsargul** [Tyrrrz/DiscordChatExporter#1238] [Q&A] [answered]
I'm using the command below to export a very large server of mine, with dozens of channels. Naturally this takes a very long time, and with a poor internet connection, it is likely I'd need to leave my computer on and running for well over a day or two to fully export everything. I don't want to leave my computer running for such an extended period of time so I was wondering if there's a way of using the command over multiple "sessions", where each sequential usage skips over the already exported data which was downloaded already. Currently when using the command, every time I run it anew, data is exported anew regardless of if it exists already or not.

`./DiscordChatExporter.Cli exportguild --token <token> -g <guild ID> --media --reuse-media true --markdown false --format Json --include-threads none --parallel 15 --output <directory>`

Any help would be appreciated

**Comment by Tyrrrz** [comment, marked as answer]
No, there's no such functionality. Your best solution in this case is to get the list of channels (the `channels` command) and then export each of them individually. You can also use date ranges to split exports by timestamps, allowing you to export each chunk of the messages separately as well.

**Reply by CharithBhat** [reply]
Okay, but with the `export` channel method, Will I be able to include all threads? Or is this an `exportguild` only feature?

**Reply by Tyrrrz** [reply]
You can include threads in the output of the `channels` command and then pass their IDs as with normal channels

**Reply by CharithBhat** [reply]
Each channel has over 1k threads. 
I'm trying to download specific channels onto a folder, along with all threads from that channel only. Looks like there is no relationship b/w a thread and its parent channel. Hence I asked.

**Reply by Tyrrrz** [reply]
Yeah there is no way to do that now

**Reply by ireadvns** [reply]
Unfortunately this feature was not implemented. I had to get around it by exporting in monthly chunks. This way, if my export fails midway I only need to delete last month's data and restart the download from there.

**Comment by Coldblackice** [comment]
How difficult would it be to implement something like this (and would it be open to PRs)?

**Comment by suitangi** [comment]
What is the filtering syntax for date-time? I know there is `before: month date` and `after: month date`. Is there a time filter?

**Reply by 96-LB** [reply]
Use the `--after` and `--before` flags instead of the filter. You can pass in a snowflake (for example, a discord message ID) like `--after 1271358184371781724` to only export messages after a particular time.

**Reply by Tyrrrz** [reply]
You can also pass a date/time stamp in any standard format
